include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3619,6 +3619,10 @@ struct OrtApi {
    *     - "73"
    *     - "75"
    *   "device_id": The ID of the device to use when setting 'htp_arch'. Defaults to "0" (for single device).
+       "disable_htp_fp16_precision": Disable the HTP_FP16 precision so that the float32 model will be inferenced with fp32 precision.
```

Comment:
i'm a little confused. the comment here is an option for disabling (default is enabled)
But the other places (tooling has enable as the option?)

Reply:
forgot to change it here.
Was using disable_htp_fp16_precision then changed to enable_htp_fp16_precision since it has bad impact for int8 QDQ models.

Reply:
i thought about it some more. i agree. explicit option for enabling fp16 is better. if user provides an fp32 model, it probably doesn't make sense to change the precision underneath them.

Reply:
Yes.
Actually more than that.
Quantized model don't need this setting. It slows down the perf for quantized model with this setting. We can go through the graph in QNN EP to see whether it's a quantized model or not. But it has cost and I don't think it worth. Because I think the major usage is quantized model so we should keep it runs as fast as possible. And float16 model is a better option comparing to inference from fp32 model with fp16 precision.